src/main/java/com/uber/jenkins/phabricator/BuildResultProcessor.java

```diff
@@ -0,0 +1,205 @@
+            return;
+        }
+
+        if (commentWithConsoleLinkOnFailure && buildResult.isWorseOrEqualTo(hudson.model.Result.UNSTABLE)) {
+            commenter.addBuildFailureMessage();
+        } else {
+            commenter.addBuildLink();
+        }
+
+        new PostCommentTask(logger, diffClient, diff.getRevisionID(false), commenter.getComment(), commentAction).run();
+    }
+
+    public boolean processHarbormaster() {
+        final boolean harbormasterSuccess = buildResult.isBetterOrEqualTo(Result.SUCCESS);
+
+        if (runHarbormaster) {
+            logger.info("harbormaster", "Sending Harbormaster BUILD_URL via PHID: " + phid);
+            Task.Result sendUriResult = new SendHarbormasterUriTask(logger, diffClient, phid, buildUrl).run();
+
+            if (sendUriResult != Task.Result.SUCCESS) {
+                logger.info(LOGGING_TAG, "Unable to send BUILD_URL to Harbormaster");
+            }
+
+            if (getCoverage() != null) {
+                logger.info(
+                        LOGGING_TAG,
+                        String.format("Publishing coverage data to Harbormaster for %d files.", getCoverage().size())
+                );
+            }
+
+            logger.info(
+                    LOGGING_TAG,
+                    String.format("Sending build result to Harbormaster with PHID %s, success: %s",
+                            phid,
+                            harbormasterSuccess
+                    )
+            );
+
+            Task.Result result = new SendHarbormasterResultTask(logger, diffClient, phid, harbormasterSuccess, getCoverage()).run();
+            if (result != Task.Result.SUCCESS) {
+                logger.info(LOGGING_TAG, "Unable to post to harbormaster");
+                return false;
+            }
+        } else {
+            logger.info("uberalls", "Harbormaster integration not enabled for this build.");
+            if (buildResult.isBetterOrEqualTo(Result.SUCCESS)) {
+                commentAction = "resign";
+            } else if (buildResult.isWorseOrEqualTo(Result.UNSTABLE)) {
+                commentAction = "reject";
+            }
+        }
+        return true;
+    }
+
+    public void processCoverage(CoverageProvider coverageProvider) {
+        if (coverageProvider == null) {
+            logger.info(LOGGING_TAG, "No coverage provider available.");
+            return;
+        }
+        Map<String, List<Integer>> lineCoverage = coverageProvider.readLineCoverage();
+        if (lineCoverage == null || lineCoverage.isEmpty()) {
+            logger.info(LOGGING_TAG, "No line coverage available to post to Harbormaster.");
+            return;
+        }
+
+        harbormasterCoverage = convertCoverage(lineCoverage);
+    }
+
+    /**
+     * Convert line coverage to the Harbormaster coverage format
+     * @return The Harbormaster-formatted coverage
+     */
+    public Map<String, String> convertCoverage(Map<String, List<Integer>> lineCoverage) {
```

Comment:
I will move it. By "make this package private" you mean remove the `public` from the `class` definition?